### ISSUE TITLE
Targets marked as up-to-date if target refers to directory.

### ISSUE THREAD

**Issue by J-Cake** [J-Cake/mkjson#2]
Targets which refer to a directory are always treated as up-to-date.

```json
"targets": {
    "build": {
        "run": "mkdir -p build"
    },
    "build/file": {
        "dependencies": ["build"],
        "run": "echo Hello > build/file"
    }
}
```
```bash
$ rm -rf build
$ pnpm exec mkjson build/file
[info]: build/file is up to date.
bash: build/file: No such file or directory
```

**Comment by J-Cake** [comment]
This makes sense because the filesystem entity representing the directory is not actually changing, but rather its contents. It might be a good idea to add an option to make targets which refer to directories to be considered _out-of-date_ if _any_ of its contents are out of date. Even better if this were the default behaviour, so opt-out rather than opt-in

**Comment by J-Cake** [comment]
Fixed by #38
